feathr_project/setup.py

```diff
@@ -20,35 +20,37 @@
     include_package_data=True,
     # consider
     install_requires=[
-        'Click',
-        "azure-storage-file-datalake>=12.5.0",
-        "azure-synapse-spark",
-        "azure-identity>=1.8.0", #fixing Azure Machine Learning authentication issue per https://stackoverflow.com/a/72262694/3193073
-        "py4j",
-        "loguru",
-        "pandas",
-        "redis",
-        "requests",
-        "pyapacheatlas",
-        "pyhocon",
-        "pandavro",
-        "pyyaml",
-        "Jinja2",
-        "tqdm",
-        "pyarrow",
+        'click<=8.1.3',
+        "py4j<=0.10.9.7",
+        "loguru<=0.6.0",
+        "pandas<=1.5.0",
+        "redis<=4.4.0",
+        "requests<=2.28.1",
+        "tqdm<=4.64.1",
+        "pyapacheatlas<=0.14.0",
+        "pyhocon<=0.3.59",
+        "pandavro<=1.7.1",
+        "pyyaml<=6.0",
+        "Jinja2<=3.1.2",
+        "pyarrow<=9.0.0",
         "pyspark>=3.1.2",
-        "python-snappy",
-        "deltalake",
-        "graphlib_backport",
+        "python-snappy<=0.6.1",
+        # fixing https://github.com/feathr-ai/feathr/issues/687
+        "deltalake<=0.5.8",
+        "graphlib_backport<=1.0.3",
         "protobuf==3.*",
-        "azure-keyvault-secrets",
-        "confluent-kafka",
-        "databricks-cli",
-        "avro",
+        "confluent-kafka<=1.9.2",
+        "databricks-cli<=0.17.3",
+        "avro<=1.11.1",
+        "azure-storage-file-datalake>=12.5.0",
```

Comment:
I tried 12.8.0 but it doesn't seem to work, maybe we need a more strict constraint here.

Reply:
good point